Mariana-21/adopt_dont_shop_2001
# Cannot Favorite A Pet More Than Once - User Story 12

### The following changes were made
_Test_
- Created test to ensure that a user cannot favorite one pet more than once
    - It is under `spec/features/favorites/destroy_spec.rb`

_View_
- Added if statement to pets show page
    - If statement checks if the pet ID is in favorites
        - If it is not it shows a link to "Add to Favorites"
        - If it is it shows a link to "Remove <Pet Name> from Favorites"
- Added favorites nav bar to the application view at top of every page

_Routes_
- Added route to destroy a pet in favorites

_Controller_
- Added `#destroy` method to favorite controller
- Added session assignment to `#update` method in favorite controller and remove quantity variable

_Model_
- Added the following methods to the favorite model
     1. `#initialize` - This method creates a new array and assigns it to a new array
     1. `#total_count` - This method returns the count of items in the `@pets` array
     1. `#add_pet(<id>)` - This method adds a pet's ID to the `@pets` array
     1. `#remove_pet(<id>)` - This method removes a pet's ID from the `@pets` array

**_Reviews Touch-Ups_**
_Test_
- Touch up review edit test to ensure that if text was removed from an edit field it returns a flash message telling the user to try again

_Controller_
- Remove render method in the review controller and replace with a redirection
- Move update ActiveRecord method above if conditional in `#update` method in reviews controller

_Routes_
- Touch up order of routes for readability

_View_
- Add section to each review
